src/reactor.cc

```diff
@@ -0,0 +1,375 @@
+  // capacities set somewhat arbitrarily
+  fuel_tracker.Init({&tritium_storage}, 1000.0);
+  blanket_tracker.Init({&blanket}, 100000.0);
+}
+
+std::string Reactor::str() {
+  return Facility::str();
+}
+
+void Reactor::Tick() {
+  if (sufficient_tritium_for_operation) {
+    SequesterTritium();
+    OperateReactor(TBR);
+    blanket_fill_policy.Start();
+    Reactor::RecordStatus("Online", fusion_power);
+  } else {
+    Reactor::RecordStatus("Shut-down", 0);
+  }
+
+
+  DecayInventory(tritium_storage);
+  DecayInventory(tritium_excess);
+  DecayInventory(tritium_sequestered);
+
+  ExtractHelium(tritium_storage);
+  ExtractHelium(tritium_excess);
+
+  if (!tritium_storage.empty() && sufficient_tritium_for_operation) {
+    double surplus = std::max(
+      tritium_storage.quantity() - reserve_inventory, 0.0);
+
+    cyclus::Material::Ptr storage_fuel = tritium_storage.Pop();
+    tritium_excess.Push(storage_fuel->ExtractQty(surplus));
+
+    if (surplus > 0.0) {
+      RecordOperationalInfo(
+          "Tritium Moved",
+          std::to_string(surplus) + "kg of T moved from storage to excess");
+    }
+    tritium_storage.Push(storage_fuel);
+    CombineInventory(tritium_excess);
+  }
+
+  // This pulls out some of the blanket each timestep so that fresh blanket can
+  // be added.
+  // NOTE: currently spent blanket dissapears into the ether. This needs to be
+  // changed in version 1 for materials tracking.
+
+  double blanket_turnover = blanket_size * blanket_turnover_rate;
+  if (!blanket.empty() && blanket.quantity() >= blanket_turnover && 
+      context()->time() % blanket_turnover_frequency == 0) {
+    cyclus::Material::Ptr spent_blanket =
+        blanket.Pop(blanket_turnover);
+    RecordOperationalInfo(
+        "Blanket Cycled",
+        std::to_string(spent_blanket->quantity()) + "kg of blanket removed");
+  } else if (!blanket.empty() &&
+             blanket.quantity() < blanket_turnover) {
+    RecordOperationalInfo(
+        "Blanket Not Cycled",
+        "Total blanket material (" + std::to_string(blanket.quantity()) +
+            ") insufficient to extract " +
+            std::to_string(blanket_turnover) + "kg!");
+  }
+}
+
+void Reactor::Tock() {
+  if (!sufficient_tritium_for_operation) {
+    try {
+      Startup();
+      fuel_startup_policy.Stop();
+      fuel_refill_policy.Start();
+    } catch (const std::exception& e) {
+      RecordOperationalInfo("Startup Error", e.what());
+      LOG(cyclus::LEV_INFO2, "Reactor") << e.what();
+    }
+  }
+
+  CombineInventory(tritium_storage);
+  CombineInventory(blanket);
+
+
+  RecordInventories(tritium_storage.quantity(), tritium_excess.quantity(), tritium_sequestered.quantity(), blanket.quantity(),
+                    helium_storage.quantity());
+}
+
+void Reactor::EnterNotify() {
+  cyclus::Facility::EnterNotify();
+
+  fuel_usage = burn_rate * (fusion_power / MW_to_GW) / seconds_per_year * context()->dt();
```

Comment:
How does your code change if this is in atoms/moles and not kg?!?!  My hunch is that it gets simpler throughout.

Reply:
Fixed. This ultimately revealed a bug, which I fixed as well.